### PR TITLE
module.c: Accept kernel modules if they are built-in

### PR DESCRIPTION
**Pull request by marcoguerri**
bd_utils_have_kernel_module checks if a kernel module is present on the
system. If the module is built-in, it is reported as not present
(issue #314).

Test Plan:
Built libblockdev and made sure that `btrfs` is visible even as a built-in module:

```
[anaconda root@host lib]# python3
Python 3.6.5 (default, Apr 23 2018, 22:53:50) 
[GCC 8.0.1 20180410 (Red Hat 8.0.1-0.21)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import gi
>>> gi.require_version("BlockDev", "2.0")
>>> 
>>> from gi.repository import BlockDev as blockdev
>>> 
>>> blockdev.utils_have_kernel_module('btrfs')
True
>>>

```

Module is built-in:
```
[anaconda root@host lib]# cat /usr/lib/modules/`uname -r`/modules.builtin | grep btrfs                                                                                                                                                                                                                                
kernel/fs/btrfs/btrfs.ko

```

### CODE REVIEW

**Comment by vojtechtrefny** [comment]
Jenkins, ok to test.

**Comment by marcoguerri** [comment]
Any insight into why the first job fails?

**Comment by vojtechtrefny** [comment]
> Any insight into why the first job fails?

LVM is broken on CentOS 7 (#368), you can ignore it.

File: src/utils/module.c [hunk 1]
```diff
@@ -70,12 +71,16 @@ gboolean bd_utils_have_kernel_module (const gchar *module_name, GError **error)
     }
 
     path = kmod_module_get_path (mod);
-    have_path = (path != NULL) && (g_strcmp0 (path, "") != 0);
+    module_present = (path != NULL) && (g_strcmp0 (path, "") != 0);
+    if (!module_present) {
+      int state = kmod_module_get_initstate(mod);
```
Comment (vojtechtrefny) [inline src/utils/module.c]: Please move the `state` definition to the start of the function (and also use `gint` instead of `int` please).

Comment (vpodzime) [reply src/utils/module.c]: Also please add space after the function name.

File: src/utils/module.c [hunk 2]
```diff
@@ -70,12 +71,16 @@ gboolean bd_utils_have_kernel_module (const gchar *module_name, GError **error)
     }
 
     path = kmod_module_get_path (mod);
-    have_path = (path != NULL) && (g_strcmp0 (path, "") != 0);
+    module_present = (path != NULL) && (g_strcmp0 (path, "") != 0);
```
Comment (vpodzime) [inline src/utils/module.c]: I like the old name of the variable better. `module_present` sounds to me like it's either loaded/available or builtin.

**Comment by marcoguerri** [comment]
God rid of `state` altogether, and left `have_path` unchanged. Still works:
```
Python 3.6.5 (default, Apr 23 2018, 22:53:50) 
[GCC 8.0.1 20180410 (Red Hat 8.0.1-0.21)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import gi
>>> gi.require_version("BlockDev", "2.0")
>>> from gi.repository import BlockDev as blockdev
>>> blockdev.utils_have_kernel_module('btrfs')
True
>>>
```

Forgot the space after the function name, though. Adding now.

**Comment by marcoguerri** [comment]
Ping :) Is it possible to have this merged?

File: src/utils/module.c [hunk 3]
```diff
@@ -71,6 +72,9 @@ gboolean bd_utils_have_kernel_module (const gchar *module_name, GError **error)
 
     path = kmod_module_get_path (mod);
     have_path = (path != NULL) && (g_strcmp0 (path, "") != 0);
+    if (!have_path) {
+      have_path = kmod_module_get_initstate (mod) == KMOD_MODULE_BUILTIN;
```
Comment (vpodzime) [inline src/utils/module.c]: I'd really rather see this as `builtin = kmod_module_get_initstate (mod) == KMOD_MODULE_BUILTIN;` with `gboolean builtin = FALSE;` added to the beginning of the function.

**Comment by vpodzime** [comment]
@marcoguerri could you please address the above nitpick so this can be merged?

**Comment by marcoguerri** [comment]
I will try by the end of the day. Sorry for the delay, got tangled up in other stuff.

**Comment by marcoguerri** [comment]
Addressed comment.

**Pull request merged**